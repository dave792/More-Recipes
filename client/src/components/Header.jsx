import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

// actions
import signUserOut from '../actions/signout';

const Header = props => (
  <header>
    <nav
      className="navbar navbar-expand-md navbar-dark fixed-top bg-dark bg-black"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        More Recipes
      </a>
      <button
        className="navbar-toggler navbar-toggler-right"
        type="button"
        data-toggle="collapse"
        data-target="#navbarNavDropdown"
        aria-controls="navbarNavDropdown"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span className="navbar-toggler-icon" />
      </button>
      <div className="collapse navbar-collapse" id="navbarNavDropdown">
        <ul className="navbar-nav ml-auto">
          <span className="sr-only">(current)</span>
          <li className="nav-item">
            <Link className="nav-link" to="/recipes">Recipes</Link>
          </li>

          {
            props.authenticated &&
            <Link className="nav-link" to="/recipe/add">Create Recipe</Link>
          }

          {
            props.authenticated &&
            <li className="nav-item dropdown">
              <a
                className="nav-link dropdown-toggle"
                href="#"
                id="navbarDropdownMenuLink"
                data-toggle="dropdown"
                aria-haspopup="true"
                aria-expanded="false"
              >
                Dashboard
              </a>
              <div
                className="dropdown-menu"
                aria-labelledby="navbarDropdownMenuLink"
              >
                <Link
                  className="nav-link"
                  to="/profile"
                >
                  My Profile
                </Link>
                <Link
                  className="nav-link"
                  to="/users/recipes"
                >
                  My Recipes
                </Link>
                <Link
                  className="nav-link"
                  to="/users/favourites"
                >
                  My Favourites
                </Link>
              </div>
            </li>
          }

          {
            props.authenticated &&
            <li className="nav-item">
              <button
                className="btn-secondary"
                style={{ width: '100px' }}
                onClick={() => { props.signUserOut(); }}
              >
                SIGN OUT
              </button>
            </li>
          }

        </ul>
      </div>
    </nav>

  </header>
);

Header.propTypes = {
  authenticated: PropTypes.bool.isRequired,
  signUserOut: PropTypes.func.isRequired
};


const mapStateToProps = state => ({
  authenticated: state.auth.isAuthenticated
});

const mapDispatchToProps = dispatch => ({
  signUserOut: () => {
    dispatch(signUserOut());
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);

