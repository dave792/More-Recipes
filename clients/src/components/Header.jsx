import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import '../css/style.css';

// The Header creates links that can be used to navigate
// between routes.

class Header extends React.Component {
  render() {
    return (
        <header>
            <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
                <a className="navbar-brand" href="#">More Recipes</a>
                <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
                  <span className="navbar-toggler-icon"></span>
                </button>
              
                <div className="collapse navbar-collapse" id="navbarSupportedContent">
                  <ul className="navbar-nav mr-auto">
                    <li className="nav-item active">
                      <Link className="nav-link" to='/'>Home</Link> 
                      <span className="sr-only">(current)</span>
                    </li>
                    <li className="nav-item">
                    <Link className="nav-link" to='/recipes'>Recipes</Link>
                    </li>
                  </ul>
                  {
                    !this.props.auth.isAuthenticated &&
                    <form className="form-inline my-2 my-lg-0">
                      <input className="form-control mr-sm-2" type="email" placeholder="Email" aria-label="Email" />
                      <input className="form-control mr-sm-2" type="password" placeholder="Password" aria-label="Password" />
                      <button className="btn btn-primary my-2 my-sm-0" type="submit">Sign In</button>
                    </form>
                  }

                  {
                    this.props.auth.isAuthenticated &&
                    <p> profile </p>
                  }
                </div>
          </nav>
      </header>
    )
  }
}

const mapStateToProps = ({auth}) => {
  return {
    auth
  };
}


export default connect(mapStateToProps)(Header);

