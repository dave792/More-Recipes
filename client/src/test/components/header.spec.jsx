import React from 'react';
import { MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import Adapter from 'enzyme-adapter-react-16';
import Header from '../../components/Header';

const mockStore = configureStore([]);
Enzyme.configure({ adapter: new Adapter() });

describe('The Header component', () => {
  it('Should mount the Header component without errors', () => {
    const store = mockStore({
      auth: {
        isAuthenticated: false
      }
    });
    const tree = shallow((
      <Header store={store} />
    ));

    expect(tree).toMatchSnapshot();
  });

  it('Should render profile link when user is authenticated', () => {
    const store = mockStore({
      auth: {
        isAuthenticated: true
      }
    });
    const tree = renderer.create((
      <MemoryRouter>
        <Header store={store} />
      </MemoryRouter>
    ));
    const linkToProfile = tree.root.findAllByProps({ to: '/profile' });

    expect(linkToProfile.length).toBe(1);

    expect(tree).toMatchSnapshot();
  });
});

