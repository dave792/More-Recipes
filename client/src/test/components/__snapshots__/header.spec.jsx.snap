// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Header component Should mount the Header component without errors 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Header)
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "authenticated": false,
      "signUserOut": [Function],
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "authenticated": false,
        "signUserOut": [Function],
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`The Header component Should render profile link when user is authenticated 1`] = `
<header>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      More Recipes
    </a>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <span
            className="sr-only"
          >
            (current)
          </span>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/recipes"
            onClick={[Function]}
          >
            Recipes
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/profile"
            onClick={[Function]}
          >
            Profile
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            onClick={[Function]}
          >
            Sign out
          </a>
        </li>
      </ul>
    </div>
  </nav>
</header>
`;
